src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -112,21 +112,28 @@ export const TradeConfirm = () => {
       // e.g sell asset AccountId, and sell asset address, and sell Txid
       // If we use the "real" (which we never get) buy Tx AccountId and address. then we'll never be able to lookup a Tx in state
       // and thus will never be able to react on the completed state
-      return serializeTxIndex(
-        sellAssetAccountId!,
-        buyTxid.toUpperCase(), // Midgard monkey patch Txid is lowercase, but we store Cosmos SDK Txs uppercase
-        fromAccountId(sellAssetAccountId!).account ?? '',
-      )
+
+      const thorOrderId = sellTradeId.toUpperCase()
+      const intoRune = isRune(trade?.buyAsset.assetId ?? '')
+      return intoRune
+        ? `${buyAssetAccountId}*${thorOrderId}*${trade?.receiveAddress}*OUT:${thorOrderId}`
+        : serializeTxIndex(
+            // this doesn't yet return the correct key due to the sellTxId/buyTxId logic described below.
+            sellAssetAccountId!,
+            buyTxid.toUpperCase(), // Midgard monkey patch Txid is lowercase, but we store Cosmos SDK Txs uppercase
```

Comment:
As discussed, this is a recurrent bug with the fact that `txHistorySlice` stores checksummed Txids/addresses everywhere, but other places use lowercased / checksummed versions interchangeably.
We should either ensure we're consistent, or alternatively as a smaller lift, consistently use a util that would produce a checksummed version when comparing addresses / Txids

Reply:
For sure. This PR intentionally leave the right hand side of this ternary untouched (though adds a comment on why it's not currently working).